pkgs/development/python-modules/glean-parser/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pytestCheckHook
+, pytest-runner
+, pythonOlder
+, setuptools-scm
+# build inputs
+, appdirs
+, click
+, diskcache
+, jinja2
+, jsonschema
+, pyyaml
+, yamllint
+}:
+
+buildPythonPackage rec {
+  pname = "glean_parser";
+  version = "4.3.1";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-wZSro1pX/50TlSfFMh71JlmXlJlONVutTDFL06tkw+s=";
+  };
+
+  propagatedBuildInputs = [
+    appdirs
+    click
+    diskcache
+    jinja2
+    jsonschema
+    pyyaml
+    yamllint
+  ];
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+    pytest-runner
```

Comment:
```suggestion
```

Reply:
Removing pytest-runner makes it fail with
> ERROR: Could not find a version that satisfies the requirement pytest-runner (from versions: none)
ERROR: No matching distribution found for pytest-runner

Reply:
That should be fixable with:
```patch
diff --git a/pkgs/development/python-modules/glean-parser/default.nix b/pkgs/development/python-modules/glean-parser/default.nix
--- a/pkgs/development/python-modules/glean-parser/default.nix
+++ b/pkgs/development/python-modules/glean-parser/default.nix
@@ -26,6 +26,10 @@ buildPythonPackage rec {
     sha256 = "sha256-wZSro1pX/50TlSfFMh71JlmXlJlONVutTDFL06tkw+s=";
   };
 
+  postPatch = ''
+    substituteInPlace setup.py --replace "pytest-runner" ""
+  '';
+
   propagatedBuildInputs = [
     appdirs
     click
@@ -41,7 +45,6 @@ buildPythonPackage rec {
 
   checkInputs = [
     pytestCheckHook
-    pytest-runner
   ];
   disabledTests = [
     # https://bugzilla.mozilla.org/show_bug.cgi?id=1741668
```

Reply:
Is it much worse to just include `pytest-runner` in the input checks, even if it's useless? At least the package script looks cleaner without `postPatch`.

Reply:
It increases the closure size and breaks cross compiling. When cross compiling, tests are not run and `checkInputs` are not available. However, since `pytest-runner` is listed in `setup_requires=setup_requirements` (inside setup.py) it would cause the entire package to fail to build.

Reply:
Ok, that makes sense, thanks!